src/Idris/Parser.hs

```diff
@@ -360,18 +354,14 @@ declExtension syn ns rules =
     ruleGroup _ _ = False
 
     extSymbol :: SSymbol -> IdrisParser (Maybe (Name, SynMatch))
-    extSymbol (Keyword n) = do fc <- reservedFC (show n)
-                               highlightP fc AnnKeyword
-                               return Nothing
+    extSymbol (Keyword n) = Nothing <$ keyword (show n)
     extSymbol (Expr n) = do tm <- expr syn
                             return $ Just (n, SynTm tm)
     extSymbol (SimpleExpr n) = do tm <- simpleExpr syn
                                   return $ Just (n, SynTm tm)
-    extSymbol (Binding n) = do (b, fc) <- name
+    extSymbol (Binding n) = do (b, fc) <- withExtent name
```

Comment:
When is `withExtent` needed?

Reply:
Only when you want the FC covering the parse.  None of the parsers return an FC anymore, but any `Parsing a` can be converted to `Parsing (a, FC)` using `withExtent`.